Support rendering non-native <meter> in vertical writing mode

nt1m opened the pull request:
#### f486be28a10145a1cbb817618080f8f3481f2e0d
<pre>
Support rendering non-native &lt;meter&gt; in vertical writing mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=248182">https://bugs.webkit.org/show_bug.cgi?id=248182</a>
rdar://102595455

Reviewed by Aditya Keerthi.

Use logical properties where needed.

Also enable &lt;meter&gt; rendering behind VerticalFormControls preference.

Native rendering will be done separately.

* LayoutTests/fast/dom/HTMLMeterElement/meter-clone-expected.txt:
* LayoutTests/fast/dom/HTMLMeterElement/meter-clone.html:
* LayoutTests/fast/dom/HTMLMeterElement/meter-element-markup-expected.txt:
* LayoutTests/fast/dom/HTMLMeterElement/meter-writing-mode-expected.html:
* LayoutTests/fast/dom/HTMLMeterElement/meter-writing-mode.html:
Adjust tests accordingly.

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-computed-style.optional-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-computed-style.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-horizontal.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-horizontal.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-vertical.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-vertical.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/w3c-import.log:
* LayoutTests/platform/glib/fast/dom/HTMLMeterElement/meter-element-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-computed-style.optional-expected.txt: Added.
Imported from WPT upstream repo.

* Source/WebCore/css/horizontalFormControls.css:
(@namespace &quot;<a href="http://www.w3.org/1999/xhtml&quot">http://www.w3.org/1999/xhtml&quot</a>;;):
* Source/WebCore/css/html.css:
(select, button, input:is([type=&quot;button&quot;], [type=&quot;submit&quot;], [type=&quot;reset&quot;], [type=&quot;file&quot;])):
(meter):
(select, button, meter, input:is([type=&quot;button&quot;], [type=&quot;submit&quot;], [type=&quot;reset&quot;], [type=&quot;file&quot;])): Deleted.
* Source/WebCore/html/HTMLMeterElement.cpp:
(WebCore::HTMLMeterElement::didElementStateChange):
* Source/WebCore/html/shadow/meterElementShadow.css:
(div#value):
(div#value.suboptimum): Remove some redundant rules covered by div#value
(div#value.even-less-good):

Canonical link: <a href="https://commits.webkit.org/262395@main">https://commits.webkit.org/262395@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/67f51dfb6a5d0fc780a8dd8646af8fa0bfe3b7ef

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1440 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1470 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1520 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2363 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/1270 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1424 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1531 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1530 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/1397 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1453 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/1324 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1306 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2203 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1334 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1290 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/1251 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1312 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1329 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2397 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1365 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1218 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1283 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1297 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/356 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1407 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/w3c-import.log:
```diff
@@ -22,6 +22,14 @@ List of files:
 /LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/color-input-appearance-none-horizontal.optional.html
 /LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/color-input-appearance-none-vertical.optional-expected-mismatch.html
 /LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/color-input-appearance-none-vertical.optional.html
+/LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-computed-style.optional.html
```
Are we okay with doing a WPT import in the same commit as a code change, or should it be done separately?

nt1m replied:
yeah, it's fine. This is only a partial import with the meter tests, there are no unrelated tests being imported, w3c-import.log reflects the current state of the tree.

nt1m committed:
Support rendering non-native <meter> in vertical writing mode
https://bugs.webkit.org/show_bug.cgi?id=248182
rdar://102595455

Reviewed by Aditya Keerthi.

Use logical properties where needed.

Also enable <meter> rendering behind VerticalFormControls preference.

Native rendering will be done separately.

* LayoutTests/fast/dom/HTMLMeterElement/meter-clone-expected.txt:
* LayoutTests/fast/dom/HTMLMeterElement/meter-clone.html:
* LayoutTests/fast/dom/HTMLMeterElement/meter-element-markup-expected.txt:
* LayoutTests/fast/dom/HTMLMeterElement/meter-writing-mode-expected.html:
* LayoutTests/fast/dom/HTMLMeterElement/meter-writing-mode.html:
Adjust tests accordingly.

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-computed-style.optional-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-computed-style.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-horizontal.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-horizontal.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-vertical.optional-expected-mismatch.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-vertical.optional.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-writing-modes/forms/w3c-import.log:
* LayoutTests/platform/glib/fast/dom/HTMLMeterElement/meter-element-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/css/css-writing-modes/forms/meter-appearance-native-computed-style.optional-expected.txt: Added.
Imported from WPT upstream repo.

* Source/WebCore/css/horizontalFormControls.css:
(@namespace "http://www.w3.org/1999/xhtml";):
* Source/WebCore/css/html.css:
(select, button, input:is([type="button"], [type="submit"], [type="reset"], [type="file"])):
(meter):
(select, button, meter, input:is([type="button"], [type="submit"], [type="reset"], [type="file"])): Deleted.
* Source/WebCore/html/HTMLMeterElement.cpp:
(WebCore::HTMLMeterElement::didElementStateChange):
* Source/WebCore/html/shadow/meterElementShadow.css:
(div#value):
(div#value.suboptimum): Remove some redundant rules covered by div#value
(div#value.even-less-good):

Canonical link: https://commits.webkit.org/262395@main
